src/gatsby/gatsby-browser.js

```diff
@@ -5,14 +5,21 @@ import "bootstrap/dist/js/bootstrap.bundle"
 import "@fortawesome/fontawesome-free/css/all.min.css"
 import "./src/styles/style.scss"
 
+import React from "react"
+import useSnowplowTracking from "./src/utils/useSnowplowTracking"
+
 export const onRouteUpdate = ({ location, prevLocation }) => {
-  if (typeof window.snowplow === 'function') {
-    window.snowplow("trackPageView");
-    // refresh snowplow link click tracking to pick up new links
-    window.snowplow("refreshLinkClickTracking");
-  }
   sessionStorage.setItem("prevPath", prevLocation ? prevLocation.pathname : null);
-};
+}
+
+const SnowplowWrapper = ({ element }) => {
+  useSnowplowTracking()
+  return element
+}
+
+export const wrapPageElement = ({ element }) => {
+  return <SnowplowWrapper element={element} />
+}
```

Comment:
Do we need to add this to the template, or is it an automatic thing in Gatsby?

Reply:
If I understand correctly, it's an automatic thing. It allows you to wrap the root element of every page: https://www.gatsbyjs.com/docs/reference/config-files/gatsby-browser/#wrapPageElement
https://www.gatsbyjs.com/docs/reference/config-files/gatsby-browser/#usage
If you add `console.log()` in the `useSnowplowTracking`, you can see it when a page is loaded.